### DISCUSSION TITLE
Show svg from Uint8List

### DISCUSSION THREAD

**Discussion by JoaoMBernardo** [zathras/jovial_svg#122] [General]
Is there a way to show svgs if I have the Uint8List for it?

**Comment by zathras** [comment]
Well, what's in the `Uint8LIst`?

If it's a string that is a valid SVG XML file, then you'd first convert the byte array (the `Uint8List`) to a string according to its character encoding, using one of the standard Dart conversion APIs.  Then, feed the string into the appropriate `ScalableImage` constructor.

**Comment by JoaoMBernardo** [comment]
The Uint8List has the actual bytes for the svg has shown here

![image](https://github.com/user-attachments/assets/4fd54d75-9737-4bb5-8de4-6828c2e49e99)

I was thinking of using ScalableImageSource.fromSvg with the bytes but can't seem to.

**Comment by zathras** [comment]
SVG is, by definition, a text XML format.  So, if by "the actual bytes for the svg" you mean a text encoding of a string containing a valid SVG XML document, then you'd first convert the byte array (the `Uint8List`) to a string according to its character encoding, using one of the standard Dart conversion APIs. Then, feed the string into the appropriate `ScalableImage` constructor.

There's a good chance the character encoding is UTF8.

**Reply by JoaoMBernardo** [reply]
Thanks! I was being dumb and trying to use the _ScalableImageWidget.fromSISource_ instead of just the widget with the fromSvgString inside.

Working fine with the switch.
